include/stdlib.h

```diff
@@ -44,6 +44,12 @@ typedef unsigned size_t;
 #define _HAVE_size_t
 #endif
 
+/* NULL pointer */
+#ifndef _HAVE_NULL
+#define NULL    ((void *) 0)
+#define _HAVE_NULL
+#endif
+
```

Comment:
This test is broken - it should test for #ifndef NULL instead

Reply:
> This test is broken - it should test for #ifndef NULL instead

Nope. The change is correct. The real problem is that some headers use

```C
#define NULL 0
```

while others use

```C
#define NULL ((void*)0)
```

This was working in 2.18 and messed up in #1826.